### ISSUE TITLE
2022 NeurIPS Driving SMARTS Competition - track 2 - Problem while generating SMARTS dataset from the Waymo Motion Open dataset

### ISSUE THREAD

**Issue by tunglm2203** [huawei-noah/SMARTS#1631]
**High Level Description**
2022 NeurIPS Driving SMARTS Competition - track 2 - Problem while generating SMARTS dataset from the Waymo Motion Open dataset

**Desired SMARTS version**
0.6.1

**Operating System**
ubuntu 18.04 (docker)

**Problems**
As mentioned in _https://github.com/huawei-noah/SMARTS/issues/1577#issuecomment-1247482020_, while generating the data from Waymo Motion Open, I faced some errors. I was trying to narrow the problem in the file `validation.tfrecord-00000-of-00150` (it's also happened in other files), and found it happened in the scenario IDs:

- 43f06b0743dc9355
- 45c6cd9309b3ce87

The error from scenario `43f06b0743dc9355`: 
```
Waiting on offline_dataset/43f06b0743dc9355/. ...
WARNING:WaymoMap:ignoring 'entry_lane' feature=147 for feature=297 as it only has a single point in its polyline.
WARNING:WaymoMap:ignoring 'entry_lane' feature=165 for feature=297 as it only has a single point in its polyline.
Process Process-1:
Traceback (most recent call last):
  File "/usr/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/cli/studio.py", line 79, in _build_single_scenario_proc
    build_single_scenario(clean, allow_offset_map, scenario, seed, click.echo)
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/sstudio/build_scenario.py", line 66, in build_single_scenario
    road_map, _ = map_spec.builder_fn(map_spec)
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/default_map_builder.py", line 148, in get_road_map
    road_map = map_class.from_spec(map_spec)
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/waymo_map.py", line 933, in from_spec
    return cls(map_spec, waymo_scenario)
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/waymo_map.py", line 124, in __init__
    self._load_from_scenario(waymo_scenario)
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/waymo_map.py", line 846, in _load_from_scenario
    self._compute_lane_intersections(composites=False)
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/waymo_map.py", line 297, in _compute_lane_intersections
    if cand_id in lane.incoming_lanes or cand_id in lane.outgoing_lanes:
  File "/opt/.venv/lib/python3.8/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/waymo_map.py", line 1185, in outgoing_lanes
    return [
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/waymo_map.py", line 1186, in <listcomp>
    self._map.lane_by_id(ol) for ol in self._lane_dict["outgoing_lane_ids"]
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/waymo_map.py", line 1560, in lane_by_id
    assert lane, f"WaymoMap got request for unknown lane_id: '{lane_id}'"
AssertionError: WaymoMap got request for unknown lane_id: '165'
WARNING:WaymoMap:ignoring 'entry_lane' feature=147 for feature=297 as it only has a single point in its polyline.
WARNING:WaymoMap:ignoring 'entry_lane' feature=165 for feature=297 as it only has a single point in its polyline.
Traceback (most recent call last):
  File "examples/traffic_histories_to_observations.py", line 389, in <module>
    recorder = ObservationRecorder(
  File "examples/traffic_histories_to_observations.py", line 67, in __init__
    self._scenario = next(scenario_iter)
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/scenario.py", line 202, in variations_for_all_scenario_roots
    agent_missions = Scenario.discover_agent_missions(
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/scenario.py", line 297, in discover_agent_missions
    road_map, _ = Scenario.build_map(scenario_root)
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/scenario.py", line 459, in build_map
    return map_spec.builder_fn(map_spec)
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/default_map_builder.py", line 148, in get_road_map
    road_map = map_class.from_spec(map_spec)
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/waymo_map.py", line 933, in from_spec
    return cls(map_spec, waymo_scenario)
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/waymo_map.py", line 124, in __init__
    self._load_from_scenario(waymo_scenario)
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/waymo_map.py", line 846, in _load_from_scenario
    self._compute_lane_intersections(composites=False)
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/waymo_map.py", line 297, in _compute_lane_intersections
    if cand_id in lane.incoming_lanes or cand_id in lane.outgoing_lanes:
  File "/opt/.venv/lib/python3.8/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/waymo_map.py", line 1185, in outgoing_lanes
    return [
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/waymo_map.py", line 1186, in <listcomp>
    self._map.lane_by_id(ol) for ol in self._lane_dict["outgoing_lane_ids"]
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/waymo_map.py", line 1560, in lane_by_id
    assert lane, f"WaymoMap got request for unknown lane_id: '{lane_id}'"
AssertionError: WaymoMap got request for unknown lane_id: '165'
```

The error from scenario `45c6cd9309b3ce87`: 

```
Waiting on offline_dataset/45c6cd9309b3ce87/. ...
Process Process-1:
Traceback (most recent call last):
  File "/usr/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/cli/studio.py", line 79, in _build_single_scenario_proc
    build_single_scenario(clean, allow_offset_map, scenario, seed, click.echo)
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/sstudio/build_scenario.py", line 75, in build_single_scenario
    road_map.to_glb(os.path.join(scenario_root, "map.glb"))
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/waymo_map.py", line 989, in to_glb
    glb = self._make_glb_from_polys()
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/waymo_map.py", line 1014, in _make_glb_from_polys
    lane_dividers = self._compute_traffic_dividers()
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/waymo_map.py", line 1034, in _compute_traffic_dividers
    assert lane_to_left
AssertionError
WARNING:ObservationRecorder:No vehicle IDs specifed. Defaulting to ego vehicle (122)
INFO:ObservationRecorder:t=0, active_vehicles=0
libEGL warning: egl: failed to create dri2 screen
libEGL warning: egl: failed to create dri2 screen
libEGL warning: egl: failed to create dri2 screen
libEGL warning: egl: failed to create dri2 screen
libEGL warning: egl: failed to create dri2 screen
libEGL warning: egl: failed to create dri2 screen
libEGL warning: egl: failed to create dri2 screen
libEGL warning: egl: failed to create dri2 screen
libEGL warning: egl: failed to create dri2 screen
libEGL warning: egl: failed to create dri2 screen
libEGL warning: egl: failed to create dri2 screen
libEGL warning: egl: failed to create dri2 screen
libEGL warning: egl: failed to create dri2 screen
libEGL warning: egl: failed to create dri2 screen
libEGL warning: egl: failed to create dri2 screen
libEGL warning: egl: failed to create dri2 screen
:loader(error): Couldn't load file offline_dataset/45c6cd9309b3ce87/map.glb: not found on model path (currently: "/root/workspace/rlbench/neurips22_comp/SMARTS:/root/workspace/rlbench/neurips22_comp/SMARTS/examples:/opt/.venv/lib/python3.8/site-packages/panda3d/etc/..:/opt/.venv/lib/python3.8/site-packages/panda3d/etc/../models")
Traceback (most recent call last):
  File "examples/traffic_histories_to_observations.py", line 396, in <module>
    recorder.collect(args.vehicles_with_sensors, headless=args.headless)
  File "examples/traffic_histories_to_observations.py", line 252, in collect
    self._record_data(
  File "examples/traffic_histories_to_observations.py", line 288, in _record_data
    self._smarts.attach_sensors_to_vehicles(self.agent_interface, {veh_id})
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/smarts.py", line 874, in attach_sensors_to_vehicles
    self._agent_manager.attach_sensors_to_vehicles(agent_interface, vehicle_ids)
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/agent_manager.py", line 665, in attach_sensors_to_vehicles
    self._vehicle_index.attach_sensors_to_vehicle(
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/utils/cache.py", line 136, in wrapper
    return func(self, *args, **kwargs)
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/vehicle_index.py", line 517, in attach_sensors_to_vehicle
    Vehicle.attach_sensors_to_vehicle(sim, vehicle, agent_interface, plan)
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/vehicle.py", line 516, in attach_sensors_to_vehicle
    if not sim.renderer:
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/smarts.py", line 904, in renderer
    self._renderer.setup(self._scenario)
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/renderer.py", line 221, in setup
    map_np = self._showbase_instance.loader.loadModel(map_path, noCache=True)
  File "/opt/.venv/lib/python3.8/site-packages/direct/showbase/Loader.py", line 298, in loadModel
    raise IOError(message)
OSError: Could not load model file(s): ['offline_dataset/45c6cd9309b3ce87/map.glb']
Exception ignored in: <function SMARTS.__del__ at 0x7fa0b5db6820>
Traceback (most recent call last):
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/smarts.py", line 856, in __del__
TypeError: 'NoneType' object is not callable
Exception ignored in: <function _ShowBaseInstance.__del__ at 0x7fa0adb905e0>
Traceback (most recent call last):
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/renderer.py", line 140, in __del__
  File "/root/workspace/rlbench/neurips22_comp/SMARTS/smarts/core/renderer.py", line 136, in destroy
  File "/opt/.venv/lib/python3.8/site-packages/direct/showbase/ShowBase.py", line 601, in destroy
TypeError: 'NoneType' object is not callable
```

Could you check this?

**Comment by Adaickalavan** [comment]
Hi @tunglm2203,

Scenarios with issues have been removed and the suggested [dataset](https://github.com/smarts-project/smarts-project.offline-datasets) has been updated.

**Comment by Adaickalavan** [comment]
We hope the above addresses your question and thus this issue is being closed. Please feel free to reopen this issue if your problem persists or open a new issue for other questions.
